course/src/main/java/org/openedx/course/presentation/unit/html/HtmlUnitViewModel.kt

```diff
@@ -0,0 +1,48 @@
+package org.openedx.course.presentation.unit.html
+
+import android.content.res.AssetManager
+import android.util.Log
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+import org.openedx.core.BaseViewModel
+import org.openedx.core.config.Config
+import org.openedx.core.extension.readAsText
+import org.openedx.core.system.AppCookieManager
+import org.openedx.core.system.connection.NetworkConnection
+import org.openedx.core.system.notifier.CourseCompletionSet
+import org.openedx.core.system.notifier.CourseNotifier
+
+class HtmlUnitViewModel(
+    private val config: Config,
+    private val edxCookieManager: AppCookieManager,
+    private val networkConnection: NetworkConnection,
+    private val notifier: CourseNotifier
+) : BaseViewModel() {
+
+    private val _injectJSList = MutableStateFlow<List<String>>(listOf())
+    val injectJSList = _injectJSList.asStateFlow()
+
+    val isOnline get() = networkConnection.isOnline()
+    val isCourseUnitProgressEnabled get() = config.isCourseUnitProgressEnabled()
+    val cookieManager get() = edxCookieManager
+
+    fun setWebPageLoaded(assets: AssetManager) {
+        if (_injectJSList.value.isNotEmpty()) return
+
+        _injectJSList.value = listOf(
+            //Injection to intercept completion state for xBlocks
+            assets.readAsText("js_injection/completions.js"),
+
+            //Injection to fix CSS issues for Survey xBlock
+            assets.readAsText("js_injection/survey_css.js")
+        )
```

Comment:
Maybe put this block or `readAsText` extension function in try-catch block.

Reply:
I don't think this is a good solution. If there are some problems with js files we should know about it while QA. Otherwise it will pass QA with issues.
What do you think?

Reply:
To think from the learner's perspective, it won't be a good approach to let the app crash over this (assuming QA missed it during the testing cycle). You can simply print the stack trace in the catch-block.

Reply:
done